From 1d77409888c36f01824711901d82bca808e4db4e Mon Sep 17 00:00:00 2001
From: Icarus Sparry <icarus.w.sparry@intel.com>
Date: Tue, 25 Apr 2017 16:42:38 -0700
Subject: [PATCH 1/3] Convert to stateless configuration

At runtime if CONFFILE (/etc/dnsmask.conf) doesn't exist then try
/usr/share/defaults/dnsmasq/dnsmasq.conf instead.

Signed-off-by: Icarus Sparry <icarus.w.sparry@intel.com>
---
 src/config.h | 8 ++++++++
 src/option.c | 7 +++++++
 2 files changed, 15 insertions(+)

diff --git a/src/config.h b/src/config.h
index 7d08f7d086d1..8d382ba61ada 100644
--- a/src/config.h
+++ b/src/config.h
@@ -218,6 +218,14 @@ RESOLVFILE
 #   endif
 #endif
 
+#ifndef DEFAULT_CONFFILE
+#   if defined(__FreeBSD__)
+#      define DEFAULT_CONFFILE "/usr/local/share/defaults/dnsmasq/dnsmasq.conf"
+#   else
+#      define DEFAULT_CONFFILE "/usr/share/defaults/dnsmasq/dnsmasq.conf"
+#   endif
+#endif
+
 #ifndef RUNFILE
 #   if defined(__ANDROID__)
 #      define RUNFILE "/data/dnsmasq.pid"
diff --git a/src/option.c b/src/option.c
index 1f698daf0452..28c10b5cab74 100644
--- a/src/option.c
+++ b/src/option.c
@@ -4986,6 +4986,13 @@ void read_opts(int argc, char **argv, char *compile_opts)
   char *buff = opt_malloc(MAXDNAME);
   int option, testmode = 0;
   char *arg, *conffile = NULL;
+#ifdef DEFAULT_CONFFILE
+  struct stat st = {0};
+  if (stat(CONFFILE, &st) != 0)
+    {
+      conffile=DEFAULT_CONFFILE;
+    }
+#endif
       
   opterr = 0;
 
-- 
2.12.2

